src/mscorlib/shared/System/ReadOnlySpan.cs

```diff
@@ -321,5 +321,45 @@ public T[] ToArray()
         /// Returns a 0-length read-only span whose base is the null pointer.
         /// </summary>
         public static ReadOnlySpan<T> Empty => default(ReadOnlySpan<T>);
+
+        /// <summary>Gets an enumerator for this span.</summary>
+        public Enumerator GetEnumerator() => new Enumerator(this);
+
+        /// <summary>Enumerates the elements of a <see cref="ReadOnlySpan{T}"/>.</summary>
+        public ref struct Enumerator
+        {
+            /// <summary>The span being enumerated.</summary>
+            private readonly ReadOnlySpan<T> _span;
+            /// <summary>The next index to yield.</summary>
+            private int _index;
+            /// <summary>The value to return from Current.</summary>
+            private T _current;
+
+            /// <summary>Initialize the enumerator.</summary>
+            /// <param name="span">The span to enumerate.</param>
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
+            internal Enumerator(ReadOnlySpan<T> span)
+            {
+                _span = span;
+                _index = 0;
+                _current = default;
+            }
+
+            /// <summary>Advances the enumerator to the next element of the span.</summary>
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
+            public bool MoveNext()
+            {
+                if (_index < _span.Length)
+                {
+                    _current = _span[_index++];
+                    return true;
+                }
+
+                return false;
+            }
+
+            /// <summary>Gets the element at the current position of the enumerator.</summary>
+            public T Current => _current;
```

Comment:
Will we want to change this to return `readonly ref T` once #14727 goes through?

Reply:
Yup.  This can be merged and then @ahsonkhan can update his PR to include that tweak.  `foreach` itself doesn't yet support being able to use a ref iteration variable, but it works fine with a Current that returns a ref, just dereferencing it.

Reply:
Actually, I changed it to be ref readonly now and just made the implementation match, so that we don't need to update the signature later; then we can simplify the implementation when the indexer is ref readonly returning.